### DISCUSSION TITLE
Lifecycle hooks

### DISCUSSION THREAD

**Discussion by emtilmont** [Cornutum/tcases#263] [Q&A] [answered]
It would be great if there was support in the `tcases-api-test` command to support setup and teardown lifecycle hooks. Since the tests that are generated from OAS run against a live server, they require live test data. Because JUNIT runs in random order by default, relying on data setup by other tests is insufficient, since the dependencies are not known to exist at test runtime.

Is it possible to support lifecycle hooks for setup and teardown such as `beforeEach` or `afterEach`?

**Comment by kerrykimbrough** [comment, marked as answer]
Another suggestion: generate tests with a specific base class, using the `-b` option (or if using Maven, `-DbaseClass`). Then define your `@Before` and `@After` methods in the base class.

**Comment by emtilmont** [comment]
@kerrykimbrough that definitely could work. i am probably just overlooking it somewhere but could you please link me to documentation about providing a base class for `tcase-api-test` generated tests?

**Comment by kerrykimbrough** [comment]
An example is shown [here](https://github.com/Cornutum/tcases/blob/master/tcases-openapi/Running-Api-Test-Cases.md#example-create-a-specific-testng-class). Among several other options, this illustrates the use and effect of `-b`.

**Comment by kerrykimbrough** [comment]
@emtilmont If you're happy with my suggestion, please mark this discussion as "Answered"
